src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,897 @@
+                For i As Integer = 0 To 6
+                    Assert.True(IO.File.Exists(FileList(i)))
+                Next
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetFiles_Directory_SearchOption_Wildcards()
+            Using TestBase As New FileIOTests
+                Dim FileList As ReadOnlyCollection(Of String) = FileSystem.GetFiles(TestBase.TestDirectory)
+                Assert.True(FileList.Count = 0)
+                Dim TestFileList As New List(Of String)
+                For i As Integer = 0 To 5
+                    TestFileList.Add(CreateTestFile(TestBase, SourceData, TestFileName:=$"NewFile{i}{If(i Mod 2 = 0, ".vb", ".cs")}"))
+                Next
+                FileList = FileSystem.GetFiles(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly, "*.vb")
+                Assert.True(FileList.Count = 3)
+                For i As Integer = 0 To 2
+                    Assert.True(TestFileList.Contains(FileList(i)))
+                Next
+                Dim NewSubDirectoryPath As String = IO.Path.Combine(TestBase.TestDirectory, "GetFiles_Directory_SearchOption_WildcardsNewSubDirectory")
+                IO.Directory.CreateDirectory(NewSubDirectoryPath)
+                TestFileList.Add(CreateTestFile(TestBase, SourceData, PathFromBase:="GetFiles_Directory_SearchOption_WildcardsNewSubDirectory", TestFileName:="NewFile.cs"))
+                FileList = FileSystem.GetFiles(TestBase.TestDirectory, SearchOption.SearchAllSubDirectories, "*.cs")
+                Assert.True(FileList.Contains(TestFileList.Last), "File in Subdirectory not found")
+                Assert.True(FileList.Count = 4, $"4 files expected, {FileList.Count} returned from FileSystem.GetFiles")
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetName_Path()
+            Using TestBase As New FileIOTests
+                Assert.Equal(FileSystem.GetName(TestBase.TestDirectory), IO.Path.GetFileName(TestBase.TestDirectory))
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetParentPath_Path()
+            Using TestBase As New FileIOTests
+                Dim TestDirectory As String = TestBase.TestDirectory
+                Assert.Equal(FileSystem.GetParentPath(TestDirectory), IO.Path.GetDirectoryName(TestDirectory.TrimEnd(IO.Path.DirectorySeparatorChar, IO.Path.AltDirectorySeparatorChar)))
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetTempFileName()
+            Dim TempFile As String = FileSystem.GetTempFileName
+            Assert.True(IO.File.Exists(TempFile))
+            Assert.True((New IO.FileInfo(TempFile)).Length = 0)
+            IO.File.Delete(TempFile)
+        End Sub
+
+        <Fact>
+        Public Shared Sub IOPathCombineTest()
+            Assert.Throws(Of System.ArgumentNullException)(Function() FileSystem.CombinePath(Nothing, "User"))
+        End Sub
```

Comment:
This is already tested in `CombinePathTest_BadBaseDirectory_RelativePath`.

Reply:
Deleted